src/jaxsim/mujoco/loaders.py

```diff
@@ -0,0 +1,480 @@
+        base_link_name: str,
+        floating_joint_name: str = "world_to_base",
+    ) -> str:
+        """"""
+
+        with tempfile.NamedTemporaryFile(mode="w+", suffix=".urdf") as urdf_file:
+
+            # Write the URDF string to a temporary file and move current position
+            # to the beginning.
+            urdf_file.write(urdf_string)
+            urdf_file.seek(0)
+
+            # Parse the MJCF string as XML (etree).
+            parser = ET.XMLParser(remove_blank_text=True)
+            tree = ET.parse(source=urdf_file, parser=parser)
+
+        root: ET._Element = tree.getroot()
+
+        if root.find(f".//joint[@name='{floating_joint_name}']") is not None:
+            msg = f"The URDF already has a floating joint '{floating_joint_name}'"
+            warnings.warn(msg)
+            return ET.tostring(root, pretty_print=True).decode()
+
+        # Create the "world" link if it doesn't exist.
+        if root.find(".//link[@name='world']") is None:
+            _ = ET.SubElement(root, "link", name="world")
+
+        # Create the floating joint.
+        world_to_base = ET.SubElement(
+            root, "joint", name=floating_joint_name, type="floating"
+        )
+
+        # Check that the base link exists.
+        if root.find(f".//link[@name='{base_link_name}']") is None:
+            raise ValueError(f"Link '{base_link_name}' not found in the URDF")
+
+        # Attach the floating joint to the base link.
+        ET.SubElement(world_to_base, "parent", link="world")
+        ET.SubElement(world_to_base, "child", link=base_link_name)
+
+        urdf_string = ET.tostring(root, pretty_print=True).decode()
+        return urdf_string
+
+    @staticmethod
+    def convert(
+        rod_model: rod.Model,
+        considered_joints: list[str] | None = None,
+    ) -> tuple[str, dict[str, Any]]:
+        """"""
+
+        # -------------------------------------
+        # Convert the model description to URDF
+        # -------------------------------------
+
+        # Consider all joints if not specified otherwise.
+        considered_joints = set(
+            considered_joints
+            if considered_joints is not None
+            else [j.name for j in rod_model.joints() if j.type != "fixed"]
+        )
+
+        # If considered joints are passed, make sure that they are all part of the model.
+        if considered_joints - set([j.name for j in rod_model.joints()]):
+            extra_joints = set(considered_joints) - set(
+                [j.name for j in rod_model.joints()]
+            )
+            msg = f"Couldn't find the following joints in the model: '{extra_joints}'"
+            raise ValueError(msg)
+
+        # Create a dictionary of joints for quick access.
+        joints_dict = {j.name: j for j in rod_model.joints()}
+
+        # Convert all the joints not considered to fixed joints.
+        for joint_name in set([j.name for j in rod_model.joints()]) - considered_joints:
```

Comment:
```suggestion
        for joint_name in {j.name for j in rod_model.joints()} - considered_joints:
```
You could directly use a set comprehension here